src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -816,6 +816,10 @@ internal void ValidateClusterTransaction(SaveChangesData result)
             if (TransactionMode != TransactionMode.ClusterWide)
                 return;
 
+            if (UseOptimisticConcurrency)
```

Comment:
You also need to do this check for every individual check that might have been configured.
We can set only a specific document to be optimistic concurrency checked.

Reply:
Done.